sw/device/silicon_creator/lib/drivers/flash_ctrl.c

```diff
@@ -81,10 +52,17 @@ rom_error_t flash_ctrl_read_start(uint32_t addr, uint32_t word_count,
       kFlashCtrlRegionInfo2 == (1u << FLASH_CTRL_CONTROL_PARTITION_SEL_BIT |
                                 2u << FLASH_CTRL_CONTROL_INFO_SEL_OFFSET),
       "Incorrect enum value for kFlashCtrlRegionInfo2");
+  static_assert(kFlashCtrlErasePage == 0u,
+                "Incorrect enum value for kFlashCtrlErasePage");
+  static_assert(kFlashCtrlEraseBank == 1u << FLASH_CTRL_CONTROL_ERASE_SEL_BIT,
+                "Incorrect enum value for kFlashCtrlEraseBank");
 
   // Set the partition.
   control_reg_val |= (uint32_t)region;
 
+  // Set the erase type.
+  control_reg_val |= (uint32_t)erase_type;
+
   // Set the number of words.
   control_reg_val = bitfield_field32_write(
       control_reg_val, FLASH_CTRL_CONTROL_NUM_FIELD, word_count);
```

Comment:
This should be `word_count - 1` until the relevant item in #3353 is resolved (please search for `num_words`). Please also see [this line](https://github.com/lowRISC/opentitan/blob/e3315cc32be531d7bbbf2a44dfd7f2af286bd559/sw/device/lib/flash_ctrl.c#L165).

Reply:
> Change num_words to be EXACTLY the number of words being processed, instead of -1, because this causes endless confusion.

That _is_ confusing, thanks for pointing this out!